tests/kernel/critical/src/Makefile

```diff
@@ -1,4 +1,3 @@
 ccflags-y += -I${ZEPHYR_BASE}/tests/include
 include $(ZEPHYR_BASE)/tests/Makefile.test
-
-obj-y = critical.o
+obj-y = main.o test_critical_context.o
```

Comment:
We do not need to split every testcase, splitting only has an advantage when you are testing different topics in one testcase and need the logical separation to make it easier to read and understand. Simple tests can be kept in one file.